### PR TITLE
Include Bioinfo contact while generating GRUS project

### PR DESCRIPTION
**Pull request by senthil10**
If any bioinfo contact was given while the order, that user will be added as one of project member along with PI.

### CODE REVIEW

File: taca_ngi_pipeline/deliver/deliver_grus.py [hunk 1]
```diff
@@ -452,41 +429,86 @@ def _create_delivery_project(self, pi_id, sensitive):
...
+        if self.binfo_email == self.pi_email:
+            logger.info("Bioinfo contact is same as PI, so will be ignored and not be used")
+        elif self.binfo_email:
+            try:
+                self.binfo_snic_id = self._get_user_sinc_id(self.binfo_email)
+                logger.info("SNIC id for bioinfo contact for project {} found: {}".format(self.projectid, self.binfo_snic_id))
+            except:
+                logger.warning("Was not able to get SNIC id for bioinfo contact with email {}".format(self.binfo_email))
+        
+    def _get_user_sinc_id(self, uemail):
```
Comment (b97pla) [inline taca_ngi_pipeline/deliver/deliver_grus.py]: typo: sinc --> snic

Comment (senthil10) [reply taca_ngi_pipeline/deliver/deliver_grus.py]: Seems like I managed to have the same typo even while calling the function 😅

File: taca_ngi_pipeline/deliver/deliver_grus.py [hunk 2]
```diff
@@ -452,41 +429,86 @@ def _create_delivery_project(self, pi_id, sensitive):
...
+                raise e
+        # try getting PI SNIC ID
+        try:
+            self.pi_snic_id = self._get_user_sinc_id(self.pi_email)
+            logger.info("SNIC PI-id for delivering of project {} is {}".format(self.projectid, self.pi_snic_id))
+        except Exception, e:
+            logger.error("Cannot fetch PI SNIC id using snic API. Error says: {}".format(str(e)))
+            raise e
+    
+    def _set_binfo_details(self):
```
Comment (Hammarn) [inline taca_ngi_pipeline/deliver/deliver_grus.py]: We were talking about having more contact people added automatically through the info the user makes in the order portal. If binfo_email was a list and we instead looped through it and tried to add all members of it then that would make it easier. 
Maybe rename `binfo_email` to something else in that case, like `extra_emails` or whatever.

**Comment by senthil10** [comment]
Valid suggestions @Hammarn :) However, (as fas as I know) currently we have only 3 contact details i.e. PI, Primary Contact and Bioinfo contact in order portal. The list of people to be added to GRUS is not yet available (not sure if it will be in future too) in order portal.

So now for the changes, I could also add the primary contact to the GRUS project along with PI and Bioinfo contact by default. Also as you mentioned it might be a good idea to add a command line parameter as well 👍

File: taca_ngi_pipeline/deliver/deliver_grus.py [hunk 3]
```diff
@@ -463,28 +462,31 @@ def _set_pi_details(self, given_pi_email=None):
...
+            binfo_email = prj_order.get('fields', {}).get('project_bx_email')
+            if binfo_email and binfo_email != self.pi_email and binfo_email not in other_member_emails:
+                other_member_emails.append(binfo_email)
         except:
-            logger.warning("Was not able to get bioinfo contact for project {}".format(self.projectid))
-        # try getting snic id for bioinfo contact
-        if self.binfo_email == self.pi_email:
-            logger.info("Bioinfo contact is same as PI, so will be ignored and not be used")
-        elif self.binfo_email:
+            pass # nothing to worry, just move on
```
Comment (b97pla) [inline taca_ngi_pipeline/deliver/deliver_grus.py]: do you expect an exception here? if so, perhaps you could be specific about which type of exception to catch. if you don't expect any exceptions, you should probably skip the try-except construct altogether

File: taca_ngi_pipeline/deliver/deliver_grus.py [hunk 4]
```diff
@@ -463,28 +462,31 @@ def _set_pi_details(self, given_pi_email=None):
...
+        self.other_member_snic_ids = []
+        # try getting appropriate contact emails
         try:
-            self.binfo_email = self._get_order_detail()['project_bx_email']
-            assert self.binfo_email #check its not empty string
-            logger.info("Bioinfo contact for project {} found: {}".format(self.projectid, self.binfo_email))
+            prj_order = self._get_order_detail()
+            owner_email = prj_order.get('owner', {}).get('email')
+            if owner_email and owner_email != self.pi_email and owner_email not in other_member_emails:
+                other_member_emails.append(owner_email)
```
Comment (b97pla) [inline taca_ngi_pipeline/deliver/deliver_grus.py]: not sure that I agree that you should add the primary contact email by default at this point (unless that is something you have decided to always do). I would suggest passing a flag on the command line or having a setting in the config that can moderate whether to include this contact or not. That way, you could still deploy the code but just skip adding this contact.

File: taca_ngi_pipeline/cli.py [hunk 5]
```diff
@@ -78,14 +78,18 @@ def deliver(ctx, deliverypath, stagingpath, uppnexid, operator, stage_only, forc
...
             help='pi-email, to be specified if PI-email stored in statusdb does not correspond SUPR PI-email')
 @click.option('--sensitive/--no-sensitive',
-            default = True,
+            default=True,
             help='flag to specify if data contained in the project is sensitive or not')
 @click.option('--hard-stage-only',
             is_flag=True,
             default = False,
             help='Perform all the delivery actions but does not run to_mover (to be used for semi-manual deliveries)')
+@click.option('--include-user-grus',
```
Comment (b97pla) [inline taca_ngi_pipeline/cli.py]: the variable name is not so intuitive imho, but that's a minor detail

**Comment by senthil10** [comment]
Tack for the comments @b97pla, I have made new commit addressing those :)

**Comment by b97pla** [comment]
:+1: looks good to me now

**Comment by Hammarn** [comment]
I'm happy as well, merging!

**Pull request merged**